### DISCUSSION TITLE
Routing section improvement

### DISCUSSION THREAD

**Discussion by EduardVasilenko** [becarpenter/book6#37] [General]
Hi Brian,
It is possible to add:
Theoretically, IPv6 prefixes are 4x longer and may consume 4x more memory.
Practically, the majority of prefixes are restricted to the first 64 bits that may need 2x more memory,
and the typically hierarchical IPv6 address summarization permits having 10x less number of prefixes in the routing table.
Hence, overall IPv6 routing and forwarding tables are more scalable thanks to the hierarchical address plan.
Eduard

**Comment by cron2** [comment]
Hi,

On Tue, Nov 22, 2022 at 06:13:25AM -0800, Eduard Vasilenko wrote:
> and the typically hierarchical IPv6 address summarization permits having 10x less number of prefixes in the routing table.

"Permit", yes, but today's numbers show ~900.000 IPv4 prefixes vs.
~150.000 IPv6 prefixes - that's just 6x less, not 10x.

I would generally avoid putting hard numbers here.  People tend to do
silly things with the global routing table, and "oh, I want to deaggregate
my /32 into 65.000 /48s, just because I can" means that the IPv6 table
could easily grow to be larger than the v4 table...

gert
-- 
"If was one thing all people took for granted, was conviction that if you 
 feed honest figures into a computer, honest figures come out. Never doubted 
 it myself till I met a computer with a sense of humor."
                             Robert A. Heinlein, The Moon is a Harsh Mistress

Gert Doering - Munich, Germany                             ***@***.***

**Comment by becarpenter** [comment]
Yes, it is worth mentioning the point but not making unsafe claims. (I could rant at length about the stupidity of assigning /48s as if there was no tomorrow, but given our failure to find a real multihoming solution, there is not much point in that rant.)

Will update the text.

**Comment by becarpenter** [comment]
Closed as done
